From 806d3d7e1a25606afc8fccd45cf82ce81bb8b245 Mon Sep 17 00:00:00 2001
From: Gareth Williams <gareth.williams.jx@renesas.com>
Date: Wed, 19 Jan 2022 11:15:34 +0000
Subject: [PATCH] Add poky cross compiler to build_aarch64_lib.sh

This includes removing the default -march setting so that yocto
can set hardware specific optimisations without conflict.

Signed-off-by: Gareth Williams <gareth.williams.jx@renesas.com>
---
 tensorflow/lite/tools/make/build_aarch64_lib.sh         | 3 +--
 tensorflow/lite/tools/make/targets/aarch64_makefile.inc | 2 --
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/tensorflow/lite/tools/make/build_aarch64_lib.sh b/tensorflow/lite/tools/make/build_aarch64_lib.sh
index 832d619..9aeea95 100755
--- a/tensorflow/lite/tools/make/build_aarch64_lib.sh
+++ b/tensorflow/lite/tools/make/build_aarch64_lib.sh
@@ -28,5 +28,4 @@ else
   NO_JOB=1
 fi
 
-make -j ${NO_JOB} TARGET=aarch64 -C "${TENSORFLOW_DIR}" -f tensorflow/lite/tools/make/Makefile $@
-
+make -j ${NO_JOB} TARGET=aarch64 AR="${AR}" CC="${CC}" CXX="${CXX}" -C "${TENSORFLOW_DIR}" -f tensorflow/lite/tools/make/Makefile $@
diff --git a/tensorflow/lite/tools/make/targets/aarch64_makefile.inc b/tensorflow/lite/tools/make/targets/aarch64_makefile.inc
index 32eab51..14171d2 100644
--- a/tensorflow/lite/tools/make/targets/aarch64_makefile.inc
+++ b/tensorflow/lite/tools/make/targets/aarch64_makefile.inc
@@ -7,13 +7,11 @@ ifeq ($(TARGET),aarch64)
   TARGET_OUT_DIR := linux_aarch64
 
   CXXFLAGS += \
-    -march=armv8-a \
     -funsafe-math-optimizations \
     -ftree-vectorize \
     -fPIC
 
   CFLAGS += \
-    -march=armv8-a \
     -funsafe-math-optimizations \
     -ftree-vectorize \
     -fPIC
-- 
2.7.4

